common/archiver/filestore/queryParser.go

```diff
@@ -102,13 +102,13 @@ func (p *queryParser) convertWhereExpr(expr sqlparser.Expr, parsedQuery *parsedQ
 		return errors.New("where expression is nil")
 	}
 
-	switch expr.(type) {
+	switch exprT := expr.(type) {
```

Comment:
you can rebind the same name
```suggestion
	switch expr := expr.(type) {
```